src/plugins/unified_histogram/public/utils/external_vis_context.ts

```diff
@@ -42,7 +42,7 @@ export const exportVisContext = (
       ? {
           suggestionType: visContext.suggestionType,
           requestData: visContext.requestData,
-          attributes: removeTablesFromLensAttributes(visContext.attributes),
+          attributes: removeTablesFromLensAttributes(visContext.attributes).attributes,
```

Comment:
Why this change? I cant follow in the code? So now the attibutes have another property attributes? 🤔

Reply:
Before the Unified Histogram was passing either `attributes` or a wrapper around `attributes` with no type check, and we had to handle in Lens. With stricter types TS noticed that the `attributes` prop on the left didn't always had the right object type, so I've refactored to always return a full `Lens*State` all the time.

Reply:
I see cc @jughosta as she knows the code here better.

Reply:
I'm not sure if this is related, but I noticed what looks to be an issue here in `lens_custom_renderer_component`: https://github.com/elastic/kibana/blob/06c715d854b8a3a5529267e7ef4801ab335b78ef/x-pack/plugins/lens/public/react_embeddable/renderer/lens_custom_renderer_component.tsx#L57-L66

When we merge those objects, `initialStateRef.current` is an object with an `attributes` prop while `props.attributes` is just the raw attributes. This caused an object containing both the raw attributes and an `attributes` prop to be passed to `updateAttributes` when I changed the filters:
![update_attributes](https://github.com/user-attachments/assets/64c24276-f28f-499b-b9e6-f0a0eec8d2ae)

I couldn't tell what this impacted but it didn't seem intentional.

Reply:
Yes thanx for checking this Davis, I remembered that detail this is why I pinged the team

Reply:
I've fixed the `attributes` nesting issue in [5386438](https://github.com/elastic/kibana/pull/186642/commits/5386438895d28a02b2263e234a873dd9e5e1c4fd)